svgnative/ports/cairo/CairoSVGRenderer.cpp

```diff
@@ -0,0 +1,473 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+#include "CairoSVGRenderer.h"
+#include "base64.h"
+#include "Config.h"
+#include "cairo.h"
+#include <math.h>
+#include "CairoImageInfo.h"
+
+namespace SVGNative
+{
+CairoSVGPath::CairoSVGPath()
+{
+    cairo_surface_t* sf = cairo_recording_surface_create ( CAIRO_CONTENT_COLOR_ALPHA, NULL );
+    mPath = cairo_create (sf);
```

Comment:
Yes, not cairo_path_t but cairo_t. This is because cairo has no API to amend existing cairo_path_t object. For example, cairo_line_to() would take cairo_t object, not cairo_path_t object. On the other hand, SVGNative::Path object has the methods to amend the path itself, like LineTo() method.
Therefore, although it looks strange, I decided to use cairo_t object for mPath.